tests/streams/test_java.py

```diff
@@ -0,0 +1,189 @@
+from src.betterproto.streams import CodedInputStream, CodedOutputStream
+from tests.output_betterproto.oneof import Test as OneofTest
+from tests.output_betterproto.nested import Test as NestedTest, TestMsg as NestedTestMsg
+
+from subprocess import run
+from shutil import which, copytree, rmtree
+from os.path import exists
+from os import makedirs
+
+import pytest
+
+
+correct_oneof = OneofTest().from_dict(
+    {"pitied": 1, "just_a_regular_field": 123456789, "bar_name": "Testing"}
+)
+
+correct_nested = NestedTest().from_dict(
+    {
+        "nested": {"count": 1},
+        "sibling": {"foo": 2},
+        "sibling2": {"foo": 3},
+        "msg": NestedTestMsg.THIS,
+    }
+)
+
+
+@pytest.fixture(scope="module")
+@pytest.mark.skipif(
+    not (
+        exists("tests/output_java/nested/NestedOuterClass.java")
+        and exists("tests/output_java/oneof/Oneof.java")
+    ),
+    "Missing Java sources in `output_java` - "
+    "you may not have a working installation of `protoc`",
+)
+@pytest.mark.skipif(which("mvn") is None, "Maven is absent and is required")
+@pytest.mark.skipif(which("jar") is None, "`jar` command is absent and is required")
+def compile_jar(output_dir):
+    # Delete any existing generated classes
+    rmtree("tests/streams/java/src/main/java/nested/", ignore_errors=True)
+    rmtree("tests/streams/java/src/main/java/oneof", ignore_errors=True)
+
+    # Grab the generated classes
+    copytree(
+        "tests/output_java/streams/nested/", "tests/streams/java/src/main/java/nested/"
+    )
+    copytree(
+        "tests/output_java/streams/oneof/", "tests/streams/java/src/main/java/oneof/"
+    )
+
+    # Compile the JAR
+    proc_maven = run(["mvn", "clean", "install", "-f", "tests/streams/java/pom.xml"])
+    if proc_maven.returncode != 0:
+        pytest.skip("Maven circle-test.jar build failed")
+
+
+@pytest.fixture(scope="module")
+def output_dir():
+    if not exists("tests/output_streams/"):
```

Comment:
Probably better to use `tempfile.TemporaryDirectory` here as well (see my other comment)

Reply:
Not sure if I want to do this, because doing it this way means that if there is an issue, a dev can look at what is actually generated to see what's wrong - let me know what you think